### ISSUE TITLE
About codes

### ISSUE THREAD

**Issue by lsrock1** [zhihou7/BatchFormer#1]
Thank you for your awesome work!

where is the utils code and do you have a plan to open the SWAD code?

https://github.com/zhihou7/BatchFormer/blob/56e93ee76440738541ae7025a84a2d9cc10b2709/domain_generalization/baseline.py#L30

**Comment by zhihou7** [comment]
Thanks for your interest.

Hi, you should first install this library (https://github.com/thuml/Transfer-Learning-Library.git).  I think you can copy the baseline.py into Transfer-Learning-Library/examples/domain_generalization/image_classification/, and run it.

If you need, I can release the code on SWAD. Different from applications on other methods, we set a smaller learning rate for the parameters of BatchFormer.

**Comment by lsrock1** [comment]
Thank you for your really fast response and helpful comment!

**Comment by zhihou7** [comment]
@lsrock1 I will update the code with SWAD in a few minutes.

**Comment by lsrock1** [comment]
Please let me know when you release the SWAD code!

**Comment by zhihou7** [comment]
I have pushed the code of swad. Actually, I have just changed files as follows,

domain_generalization/swad/README.md
domain_generalization/swad/domainbed/trainer.py
domain_generalization/swad/domainbed/algorithms/algorithms.py
domain_generalization/swad/train_all.py

Btw, empirically, BatchFormer might overfit the pattern of sample relationships in some cases. Therefore, we reduce the learning rate or increase weight decay for BatchFormer modules in some cases. Usually, this is not neccessary.
